t/unittest/UnifiedApi/Controller/Violations.t

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/perl
+
+=head1 NAME
+
+Violations
+
+=cut
+
+=head1 DESCRIPTION
+
+unit test for VioLations
+
+=cut
+
+use strict;
+use warnings;
+use DateTime;
+use lib '/usr/local/pf/lib';
+use pf::dal::violation;
+use pf::violation;
+
+BEGIN {
+    #include test libs
+    use lib qw(/usr/local/pf/t);
+    #Module for overriding configuration paths
+    use setup_test_config;
+}
+
+use Test::More tests => 24;
+use Test::Mojo;
+use Test::NoWarnings;
+my $t = Test::Mojo->new('pf::UnifiedApi');
+
+#truncate the violation table
+pf::dal::violation->remove_items();
+
+#run unittest
+$t->get_ok('/api/v1/violations' => json => { })
+  ->json_is('/',undef)
+  ->status_is(200);
+
+#setup
+my $mac = '00:01:02:03:04:05';
+my $tenant_id = 1;
+my $vid = 1300000; #'Generic' Violation
+
+#clear node, violation will auto-insert
+#my $node_status = pf::node::node_delete($mac, $tenant_id);
+
+#insert known data (violation)
+my $dt_format = DateTime::Format::Strptime->new(pattern => '%Y-%m-%d %H:%M:%S');
+my $dt_start = DateTime->now(time_zone=>'local');
+my $dt_release = DateTime->now(time_zone=>'local')->add(seconds => 7200);
+my $violation_status = pf::violation::violation_add($mac, $vid, (
```

Comment:
Posting to /violations should do exactly this behavior, so use it instead of calling pf::violation::violation_add